application/src/main/kotlin/net/corda/v5/application/crypto/DigitalSignatureVerificationService.kt

```diff
@@ -13,17 +15,36 @@ import java.security.PublicKey
 @DoNotImplement
 interface DigitalSignatureVerificationService {
     /**
-     * Verifies a digital signature by using [signatureSpec].
-     * Always throws an exception if verification fails.
      *
-     * @param publicKey The signer's [PublicKey].
-     * @param signatureSpec The signature spec.
-     * @param signatureData The signatureData on a message.
-     * @param clearData The clear data/message that was signed (usually the Merkle root).
+     * Check a payload and signature, and optionally a digest algorithm, check the signature matches using a specific public key.
+     * This overload is preferable so that calling code does not need to assume specific cryptographic algorithms, since
+     * the best choices change over time.
+     *
+     * @param payload The payload/message that may be in clear data that was signed (e.g. the Merkle root).
+     * @param signatureData The signature data and public key, which should have been earlier produced by [SigningService.sign].
+     * @param publicKey A verified public key aginst which to verify the signature.
+     * @return The metadata associated with the signature, passed in to [SigningService.sign]
+     *
+     * @throws CryptoSignatureException If verification of the digital signature fails.
+     * @throws IllegalArgumentException If the signature scheme is not supported or if any of the clear or signature
+     * data is empty.
+     */
+    fun verify(payload: ByteArray, signatureData: DigitalSignature.WithMetadata, publicKey: PublicKey, digestAlgorithmName: DigestAlgorithmName?): Map<String, String>
```

Comment:
If we put the SignatureSpec onto the metadata before signing then there is no need to pass in the digest name and the return is not needed either as it is in the object